### DISCUSSION TITLE
Required Searchable Select Field in Custom Pages Bug

### DISCUSSION THREAD

**Discussion by griggsk** [filamentphp/filament#12467] [Help]
### Package

Form builder

### Package Version

v3

### How can we help you?
When using a select field with searchable and required in a Livewire custom page the required validation fails. Removing the searchable and it works. It also works fine in a resource panel.

**Code**

`Forms\Components\Select::make('zipcode_id')
                    ->relationship('zipcode', 'zipcode')
                    ->label('Your Oregon Zipcode')
                    ->required()
                    ->searchable(),`

In the custom page, I can search and it fills the field with the value but is still fails the required validation so doesn't save. 
I get the required error message. 

![Screen Shot 2024-04-23 at 11 27 22 PM](https://github.com/filamentphp/filament/assets/183902/e8164de0-1af2-4626-a362-ae9dd2d10c00)

**Console errors**
When selecting: 

- Uncaught TypeError: this.select is null
- Uncaught (in promise) TypeError: this.select is null

Any ideas on how to fix? Thanks

**Comment by VincvanAs** [comment]
I just had a similar issue with the searchable select component. I don't know if this applies to your error as well, but maybe it will help you.

A temporary fix for you would be to check the path of your form state.

```
Form::make()
    ->statePath('data')
```

And based on this add this to the mount:

```
public function mount()
{
    $this->data = [
        'zipcode_id' => null,
    ];
}
```

**Reply by Fanom2813** [reply]
this works , thanks

**Comment by griggsk** [comment]
Thanks. That didn't work for me. 

When I remove the required validation the zipcode does save correctly so it seems to be a conflict with the required validation.

**Reply by leandrocfe** [reply]
Could you share the whole code please?

**Comment by griggsk** [comment]
Absolutely: https://github.com/griggsk/osp/blob/main/CreateChild.php

I did more testing and it is not saving the zip to the database when I remove required. But it does save when I remove searchable.

**Reply by griggsk** [reply]
Line 92 is where the zip is

**Comment by lordarrakis** [comment]
Why is there no better answer for this? I'm getting this bug a lot in my app

**Comment by KevinWinardi** [comment]
Based on the documentation in this [link](https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component#adding-the-form), you have to write code `$this->form->fill()` in function mount().
```
public function mount(): void
{
    $this->form->fill();
}
```
I also had a similar problem, apparently with that code it worked.
